src/psyclone/dynamo0p3.py

```diff
@@ -1017,12 +1017,38 @@ class DynamoPSy(PSy):
     def __init__(self, invoke_info):
         PSy.__init__(self, invoke_info)
         self._invokes = DynamoInvokes(invoke_info.calls, self)
-        # Initialise the dictionary that holds the names of required
-        # LFRic data structures and their proxies for the "use"
-        # statements in modules that contain PSy-layer routines.
-        infmod_list = ["field_mod", "integer_field_mod", "operator_mod"]
-        self._infrastructure_modules = OrderedDict(
-            (k, set()) for k in infmod_list)
+        # Initialise dictionaries for infrastructure modules
+        self._initialise_infrastructure_modules()
+
+    def _initialise_infrastructure_modules(self):
```

Comment:
I don't see why all this additional code is needed? As far as I can see, it doesn't allow us to do anything we couldn't already do and significantly adds to the line count for this file.

Reply:
I did this to preserve the order of the `use` statements now when I need to update the list of kinds as well with `use constants_mod`. However, I figured out that just adding `"constants_mod"` at the beginning of `infmod_list` produces the same effect. This is now simplified and the code changes with using sub-dictionaries of `infrastructure_modules` are reverted.

Reply:
Great.